crates/lumina_server/src/game.rs

```diff
@@ -97,6 +98,95 @@ fn respawn_spaceships(
     }
 }
 
+#[derive(Event, Debug)]
+pub struct ResetSpaceships;
+
+/// Resets all spaceship health, energy, weapon and abilities when a game starts
+fn reset_spaceships(
+    trigger: Trigger<ResetSpaceships>,
+    mut commands: Commands,
+    q_lobbies: Query<&Lobby>,
+    mut q_spaceships: Query<
+        (
+            &mut Health,
+            &MaxHealth,
+            &mut Energy,
+            &Spaceship,
+            &PlayerId,
+            Entity,
+            Option<&mut AbilityCooldown>,
+            Option<&ShadowAbilityConfig>,
+            Option<&HealAbilityConfig>,
+        ),
+        With<SourceEntity>,
+    >,
+    q_dash_cooldowns: Query<Entity, With<DashCooldown>>,
+    player_infos: Res<PlayerInfos>,
+    mut q_weapons: Query<(&mut WeaponState, &Weapon), With<SourceEntity>>,
+) {
+    let lobby_entity = trigger.entity();
+
+    // Get the Lobby component for this specific room
+    let Ok(lobby) = q_lobbies.get(lobby_entity) else {
+        warn!("No lobby found for entity {:?}", lobby_entity);
+        return;
+    };
+
+    info!("Resetting spaceships for lobby {:?}", lobby_entity);
+
+    for client_id in lobby.iter() {
+        if let Some(spaceship_entity) =
+            player_infos[PlayerInfoType::Spaceship].get(&PlayerId(*client_id))
+        {
+            if let Ok((
+                mut health,
+                max_health,
+                mut energy,
+                spaceship,
+                _player_id,
+                entity,
+                cooldown,
+                shadow_config,
+                heal_config,
+            )) = q_spaceships.get_mut(*spaceship_entity)
+            {
+                // Reset health
+                **health = **max_health;
+
+                // Reset energy
+                energy.energy = spaceship.energy.max_energy;
+                energy.cooldown = 0.0;
+
+                // Reset ability cooldown
+                if let Some(mut cooldown) = cooldown {
+                    if shadow_config.is_some() || heal_config.is_some() {
+                        cooldown.reset();
+                    }
```

Comment:
```suggestion
                // Speed up cooldown.
                let duration = cooldown.duration();
                cooldown.set_elapsed(duration);
```
Needs to speed up the timer instead of resetting the timer.